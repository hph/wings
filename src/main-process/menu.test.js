import { Menu } from 'electron';

import setApplicationMenu, { template } from './menu';

jest.mock('electron', () => ({
  Menu: {
    buildFromTemplate: jest.fn(() => 'built template'),
    setApplicationMenu: jest.fn(),
  },
}));

describe('menu', () => {
  test('template should have a static value', () => {
    expect(template).toMatchSnapshot();
  });

  test('default export should build create and set the menu', () => {
    setApplicationMenu();
    expect(Menu.buildFromTemplate).toHaveBeenCalledWith(template);
    expect(Menu.setApplicationMenu).toHaveBeenCalledWith('built template');
  });
});
