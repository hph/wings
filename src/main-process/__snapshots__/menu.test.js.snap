// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu template should have a static value 1`] = `
Array [
  Object {
    "label": "Edit",
    "submenu": Array [
      Object {
        "role": "copy",
      },
      Object {
        "role": "paste",
      },
    ],
  },
  Object {
    "label": "View",
    "submenu": Array [
      Object {
        "role": "reload",
      },
      Object {
        "role": "toggledevtools",
      },
      Object {
        "role": "togglefullscreen",
      },
    ],
  },
]
`;
